packages/dev/core/src/Materials/Node/Blocks/Dual/sceneDepthBlock.ts

```diff
@@ -25,10 +25,42 @@ export class SceneDepthBlock extends NodeMaterialBlock {
      * Defines if the depth renderer should be setup in non linear mode
      */
     @editableInPropertyPage("Use non linear depth", PropertyTypeForEdition.Boolean, "ADVANCED", {
-        notifiers: { activatePreviewCommand: true, callback: (scene) => scene.disableDepthRenderer() },
+        notifiers: {
+            activatePreviewCommand: true,
+            callback: (scene, block) => {
+                const sceneDepthBlock = block as SceneDepthBlock;
+                let retVal = false;
+                if (sceneDepthBlock.useNonLinearDepth) {
+                    sceneDepthBlock.storeCameraSpaceZ = false;
+                    retVal = true;
+                }
+                scene.disableDepthRenderer();
+                return retVal;
+            },
+        },
     })
     public useNonLinearDepth = false;
 
+    /**
+     * Defines if the depth renderer should be setup in camera space Z mode (if set, useNonLinearDepth has no effect)
+     */
+    @editableInPropertyPage("Store Camera space Z", PropertyTypeForEdition.Boolean, "ADVANCED", {
+        notifiers: {
+            activatePreviewCommand: true,
+            callback: (scene, block) => {
+                const sceneDepthBlock = block as SceneDepthBlock;
+                let retVal = false;
+                if (sceneDepthBlock.storeCameraSpaceZ) {
+                    sceneDepthBlock.useNonLinearDepth = false;
+                    retVal = true;
+                }
+                scene.disableDepthRenderer();
+                return retVal;
+            },
+        },
+    })
+    public storeCameraSpaceZ = false;

```

Comment:
should this be serialized as well ?

Reply:
Yes, it is serialized / parsed in `serialize` / `_deserialize`.